codalab/apps/teams/forms.py

```diff
@@ -89,3 +95,144 @@ def clean(self):
         cleaned_data = super(TeamMembershipForm, self).clean()
 
         return cleaned_data
+
+
+class OrganizerTeamForm(forms.ModelForm):
+
+    text_members = forms.CharField(required=False, widget=forms.Textarea)
+
+    class Meta:
+        model = Team
+        fields = ('description', 'name')
+
+    def __init__(self, *args, **kwargs):
+        self.user_list = []
+        self.competition = Competition.objects.get(pk=kwargs.pop('competition_pk'))
+        self.creator = ClUser.objects.get(pk=kwargs.pop('creator_pk'))
+        super(OrganizerTeamForm, self).__init__(*args, **kwargs)
+        # This check is to see if we have an existing object with a PK, or if we're making a new one
+        # If we have an existing we grab existing members for the form.
+        if hasattr(self, 'instance'):
+            if hasattr(self.instance, 'pk'):
+                if self.instance and self.instance.pk:
+                    team = Team.objects.get(pk=self.instance.pk)
+                    current_members = team.members.all()
+                    name_text_list = ""
+                    name_list = []
+                    for member in current_members:
+                        name_list.append(member.username)
+                        name_text_list = ",".join(name_list)
+                    self.initial['text_members'] = name_text_list
+
+    def clean_text_members(self):
+        if self.cleaned_data.get('text_members'):
+            team_members_text = self.cleaned_data.get('text_members')
+            team_members_list = team_members_text.replace(' ','').split(',')
+            # team_members_list = team_members_text.replace(' ','').split('\n')
+            self.user_list = ClUser.objects.filter(Q(username__in=team_members_list) | Q(email__in=team_members_list))
```

Comment:
i don't think we should be doing it like this, jumping from function to function kinda?

Reply:
Probably should just do all of this in the [clean()](https://docs.djangoproject.com/en/1.7/ref/forms/validation/#cleaning-and-validating-fields-that-depend-on-each-other) method

Reply:
Hmm seems nice having it all in field specific clean methods though.